File: tests/omp/async_target1.ml
open Optitrust
open Target


let _ = Flags.dump_ast_details := true

let _ = Run.script_cpp (fun _ ->

  
  !! Omp.declare_target [cFunDef "F"];
  !! Omp.end_declare_target [cFunDef "F"];
  !! Omp.task ~clause:[Shared ["z"]] [cFor "i"];
  !! Omp.target_teams ~clause:[Map_c (From, ["Z[C:CHUNKSZ]"])] [cFor "i"];
  !! Omp.parallel_for [cFor "i"];
  !! Omp.taskwait [cFor "i"];
)

File: tests/omp/async_target1.cpp
float F(float);
const int N = 1000000000;
const int CHUNKSZ = 1000000;
void init(float *, int);
float Z[N];
void pipedF(){
   int C;
   init(Z, N);
   for (C=0; C<N; C+=CHUNKSZ){
      for (int i=0; i<CHUNKSZ; i++) Z[i] = F(Z[i]);
   }
}
